src/Microsoft.PowerShell.Commands.Utility/commands/utility/MatchString.cs

```diff
@@ -248,6 +291,74 @@ public string ToString(string directory)
             return string.Join(System.Environment.NewLine, lines.ToArray());
         }
 
+        /// <summary>
+        /// Returns the string representation of the match object same format as ToString()
+        /// and adds color to the matched text if virtual terminal is supported.
+        /// </summary>
+        /// <param name="directory">Directory to use as the root when calculating the relative path.</param>
+        /// <returns>The colored string representation of the match object.</returns>
+        public string ToEmphasizedString(string directory)
+        {
+            if (!Emphasize || ! _supportsVirtualTerminal) {
+                return ToString(directory);
+            }
+
+            string originalLine = Line;
+            string open = "\u001b[7m";
+            string close = "\u001b[0m";
+
+            Line = string.Create((_matchIndexes.Count * (open.Length + close.Length)) + Line.Length, Line, (chars, buf) =>
+            {
+                int lineIndex = 0;
+                int charsIndex = 0;
+                for (int i = 0; i < _matchIndexes.Count; i++) 
+                {
+                    // Adds characters before match
+                    while (lineIndex < _matchIndexes[i])
```

Comment:
```suggestion
                    Line.CopyTo(lineIndex, chars, charsIndex, _matchIndexes[i] - lineIndex);
                    charsIndex += _matchIndexes[i] - lineIndex;
                    lineIndex = _matchIndexes[i];
```

Reply:
I'm getting "cannot convert from 'System.Span<char>' to 'char[]'" when I use CopyTo.

Reply:
@rjmholt  Can you respond to @derek-xia

Reply:
Ah I looked for this comment but couldn't find it (got a notification but it didn't take me to the comment).

This `string.Create` method is fairly obscure -- the documentation on it isn't great.

I'd go with array creation; .NET will probably be somewhat faster using `CopyTo` rather than while loops

Reply:
CopyTo() works for Span-s too https://docs.microsoft.com/en-us/dotnet/api/system.readonlyspan-1.copyto?view=netstandard-2.1

like
```c#
Line.AsSpan.Slice(...).CopyTo(chars.Slice(...));
```

Reply:
Ah I searched for that -- didn't think to look for properties. My problem with `string.Create(...)` here is that it creates the variable `buf` and I don't know what the type is

Reply:
Really Line is mapped to buf in the context.
Direction is that .Net Core is more and more optimized for Span and ReadOnlySpan. The fact is that PowerShell Core consumes more memory than Windows PowerShell. So we would use new patterns everywhere.

Reply:
Ah that's right -- `buf` if the span of the old string and `chars` is the span of the new one.

But that's unclear at best. As much as I agree about using spans over strings, that API is very poorly documented ([`String.Create`](https://docs.microsoft.com/en-us/dotnet/api/system.string.create?view=netcore-3.0#System_String_Create__1_System_Int32___0_System_Buffers_SpanAction_System_Char___0__) and the needed [`SpanAction<T, TArg>`](https://docs.microsoft.com/en-us/dotnet/api/system.buffers.spanaction-2?view=netcore-3.0) doc &mdash; the only example usage I've found is [in a blog post](https://msdn.microsoft.com/en-us/magazine/mt814808.aspx), so I feel .NET could endorse this direction a little bit better) and in this case I don't think it really buys us all that much. Plus we allocate a closure.

We can always refactor it later.

Reply:
Best docs is Core itself https://source.dot.net/#System.Private.CoreLib/shared/System/String.cs,7dd9a20e8a84bf21,references